pandas/core/indexes/api.py

```diff
@@ -101,7 +101,7 @@ def conv(i):
 
 
 def _sanitize_and_check(indexes):
-    kinds = list(set([type(index) for index in indexes]))
+    kinds = list(set(type(index) for index in indexes))
```

Comment:
this could be a set comprehension
``{type(index) for index in indexes}``

prob more of these as well